### PR TITLE
Updating to stripe/stripe-php=~2.0

### PR DESCRIPTION
**Pull request by ellisthedev**
Reopening on the `master` branch. Refer to the prior pull request for details: https://github.com/laravel/cashier/pull/152

### CODE REVIEW

**Comment by ellisthedev** [comment]
This version may be breaking now that I look at the fact you allow passing `$apiKey` to methods when the parent class expects an array of options. However, this part is untested. I'm doing a lot of work with subscriptions and one off charges this week so I will test this code in depth and report back if this needs more work.

@taylorotwell also notified me on Twitter that he has a test Stripe application that he's going to test this with as well. So if he finds anything himself I'm sure we'll hear about it.

File: src/Laravel/Cashier/Invoice.php [hunk 1]
```diff
@@ -171,7 +171,7 @@ public function discount()
     {
         setlocale(LC_MONETARY, $this->billable->getCurrencyLocale());
 
-        return round(money_format('%i', ($this->subtotal / 100) - ($this->total / 100)), 2);
+        return round(money_format('%!i', ($this->subtotal / 100) - ($this->total / 100)), 2);
```
Comment (taylorotwell) [inline src/Laravel/Cashier/Invoice.php]: What does this do?

Comment (ellisthedev) [inline src/Laravel/Cashier/Invoice.php]: So this change was to fix an error that was occurring on the unit tests. The original method the `money_format` call was returning `USD100.00`. Then when passed into round, round was returning `0`; thus breaking the unit test as `0 !== 100`.

Adding the bang keeps money_format from returning `USD` in the string.

**Comment by SylarRuby** [comment]
@taylorotwell after I ran composer update Im getting this for new users to my website: Could not determine which URL to request: Laravel\Cashier\Customer instance has invalid ID: Was there an upgrade?

**Comment by taylorotwell** [comment]
Which version of cashier are you on. This hasn't been merged yet.

**Comment by SylarRuby** [comment]
Hey @taylorotwell It a pleasure lol. The when I place this in my controller: $invoices = $user->invoices(); I get the error when a new user registers.

**Comment by ellisthedev** [comment]
This sounds like a bug in your application. That error is you trying to call Stripe with a cus_ ID that does not exist in Stripe.

Have you verified that ID exists?

**Comment by SylarRuby** [comment]
Ahhhh but how to call that method? It works ok but if a new user registers in my website it throws an error. I cant even wrap it around an if statement.

**Pull request merged**